lib/iris/_merge.py

```diff
@@ -1068,6 +1069,27 @@ def derive_space(groups, relation_matrix, positions, function_matrix=None):
     return space
 
 
+def _multidim_daskstack(stack):
```

Comment:
I think this needs more testing.
Probably a purpose-built unit test.
Could be deferred to another ticket for now, though, if it is working for all existing tests.